src/components/VirtualKeyboardKey.vue

```diff
@@ -14,88 +12,11 @@ const props = defineProps<{
 }>()
 
 const active = ref(false)
-
 const light = computed(() => new Values(props.color).getBrightness() > 50)
-
-const emit = defineEmits(['press', 'unpress'])
-
-let noteOff: NoteOff | null = null
-
-function start() {
-  active.value = true
-  if (noteOff !== null) {
-    noteOff()
-  }
-  noteOff = props.noteOn()
-}
-
-function end() {
-  active.value = false
-  if (noteOff !== null) {
-    noteOff()
-    noteOff = null
-  }
-}
-
-function onTouchStart(event: TouchEvent) {
-  event.preventDefault()
-  start()
-}
-
-function onTouchEnd(event: TouchEvent) {
-  event.preventDefault()
-  end()
-}
-
-function onMouseDown(event: MouseEvent) {
-  if (event.button !== LEFT_MOUSE_BTN) {
-    return
-  }
-  event.preventDefault()
-  emit('press')
-  start()
-}
-
-function onMouseUp(event: MouseEvent) {
-  if (event.button !== LEFT_MOUSE_BTN) {
-    return
-  }
-  event.preventDefault()
-}
-
-function onWindowMouseUp(event: MouseEvent) {
-  if (event.button !== LEFT_MOUSE_BTN) {
-    return
-  }
-  emit('unpress')
-  end()
-}
-
-function onMouseEnter(event: MouseEvent) {
-  if (!props.isMousePressed) {
-    return
-  }
-  event.preventDefault()
-  start()
-}
-
-function onMouseLeave(event: MouseEvent) {
-  if (!props.isMousePressed) {
-    return
-  }
-  event.preventDefault()
-  end()
-}
-
-onMounted(() => {
-  window.addEventListener('mouseup', onWindowMouseUp)
-})
+const playNote = usePlayNote(props.noteOn);
```

Comment:
All the play note logic is now refactored inside this hook. It is referenced here and other components.